cfgov/filing_instruction_guide/models.py

```diff
@@ -169,16 +69,47 @@ class FIGContentPage(CFGOVPage):
         ]
     )
 
+    def get_toc_headers(self, request):
+        toc_headers = []
+        parent = None
+        for section in self.content:
+            header = section.value.get("header")
+            id = section.value.get("section_id")
+            if section.block_type == "Fig_Section":
+                if parent:
+                    toc_headers.append(parent)
+                parent = {"header": header, "id": id, "children": []}
+            elif section.block_type == "Fig_Sub_Section":
+                # if the first block is a subsection instead of a section
+                if not parent:
+                    parent = {"header": "", "id": "", "children": []}
+                parent["children"].append({"header": header, "id": id})
+        toc_headers.append(parent)
+        return toc_headers
+
+    def assign_section_ids(self):
+        ind = sub_ind = sub3_ind = 0
+        for section in self.content:
+            id = ""
+            sec_type = section.block_type
+            if sec_type == "Fig_Section":
+                ind += 1
+                sub_ind = 0
+                sub3_ind = 0
+                id = f"{ind}."
+            if sec_type == "Fig_Sub_Section":
+                sub_ind += 1
+                sub3_ind = 0
+                id = f"{ind}.{sub_ind}."
+            if sec_type == "Fig_Sub_3_Section":
+                sub3_ind += 1
+                id = f"{ind}.{sub_ind}.{sub3_ind}."
+            section.value["section_id"] = id
+
+    base_form_class = FIGPageForm
     template = "filing_instruction_guide/index.html"
 
-    objects = PageManager()
-
     def get_context(self, request, *args, **kwargs):
         context = super().get_context(request, *args, **kwargs)
-        context.update(
-            {
-                "report_sections": get_report_sections(request, self),
-                "report_appendices": get_report_appendices(request, self),
-            }
-        )
+        context.update({"get_sections": self.get_toc_headers(request)})
```

Comment:
Can you rename this to just `sections`? Feels like `get_sections` implies it's a function that needs to be called

Reply:
Good call. Done in 94e6733bc591a957308339e34173f68bc206de3f.